business/workloads.go

```diff
@@ -2034,11 +2124,21 @@ func (in *WorkloadService) streamParsedLogs(cluster, namespace, name string, opt
 			break
 		}
 
-		entry := parseLogLine(line, opts.IsProxy, engardeParser)
+		var entry *LogEntry
+		if ambient {
+			entry = parseZtunnelLine(line)
+		} else {
+			entry = parseLogLine(line, opts.IsProxy, engardeParser)
+		}
+
 		if entry == nil {
 			continue
 		}
 
+		if opts.filter != nil && !filterMatches(entry.Message, opts.filter) {
+			continue
+		}
```

Comment:
We've already parsed out the entry. It might be better to filter by entry fields now like `entry.AccessLog.UpstreamCluster`. I worry we'll get false positives doing simple string matching.

You could have one app named `productpage` and another named `longerproductpage` and they both would match doing simple string matching on the whole line.

Reply:
Added a new filter